model/location_credential.rb

```diff
@@ -9,20 +9,29 @@ class LocationCredential < Sequel::Model
   many_to_one :project
   many_to_one :location, key: :id
 
+  def credentials
+    @credentials ||= if !assume_role.empty?
+      Aws::AssumeRoleCredentials.new(role_arn: assume_role, role_session_name: ENV["HOSTNAME"] || "ubi")
+    else
+      Aws::Credentials.new(access_key_id: access_key, secret_access_key: secret_key)
+    end
+  end
+
   def client
-    Aws::EC2::Client.new(access_key_id: access_key, secret_access_key: secret_key, region: location.name)
+    Aws::EC2::Client.new(region: location.name, credentials: credentials)
```

Comment:
For new code, we prefer hash value omission (also applies to next method, and maybe additional methods in the PR):

```suggestion
    Aws::EC2::Client.new(region: location.name, credentials:)
```

Reply:
I'm surprised there's not a rubocop rewrite for this...yet?